lib/routes/utils/request_helper.js

```diff
@@ -9,9 +9,52 @@
  * @returns {boolean}
  */
 function wantsKeys (request) {
-  return request.query.keys === 'true'
+  return request.query.keys === 'true' || request.query.keys === true
+}
+
+/**
+ * Returns whether or not to perform a signin verification email.
+ *
+ * @param account
+ * @param config
+ * @param request
+ * @returns {boolean}
+ */
+function shouldEnableSigninConfirmation(account, config, request) {
+
+  var confirmLogin = config.signinConfirmation && config.signinConfirmation.enabled
+  if (confirmLogin) {
+
+    var uid = account.uid.toString('hex')
+    var email = account.email
+
+    // If feature enabled, always enable for these emails
+    var emailDomain = account.email.substring(email.indexOf('@'), email.length).toLocaleLowerCase()
+    var isValidEmail = config.signinConfirmation.allowEmails.indexOf(emailDomain) > -1
```

Comment:
Ah, interesting, when I read the name "allowEmails" I thought it meant "only allow it for these emails", not "always enable it for these emails".  We sometimes use the word "force" in this sort of context, e.g. using `forceDeviceList=true` to show the device list UI.  So I wonder if something like "supportedClients" and "forcedEmails" might make it a little clearer what these config values mean.